src/read.jl

```diff
@@ -261,3 +255,197 @@ function read_demand_flexibility(filepath, interval)::DemandFlexibility
+            zone_cols_idx = findall(x -> x in flexible_zone_id, zone_list)
+
+            # for flex amt, the zone numbers are the total flexibility in the zone
+            if field == "flex_amt_up" || field == "flex_amt_dn"
+                # if input contains bus columns
+                if flexible_bus_num > 0
+                    # find if the flexibility of any bus is also specified in a zone column
+                    bus_cols_zone_id = case.bus_zone[findall(
+                        x -> x in flexible_bus_id, case.busid
+                    )]
+                    bus_cols_zone_idx = [
+                        findfirst(isequal(i), flexible_zone_id) for i in bus_cols_zone_id
+                    ]
+
+                    # substract individual bus columns from zone total flexibility
+                    for i in 1:flexible_bus_num
+                        # substract from total if the zone of this bus is specified in a zone column
+                        if !isnothing(bus_cols_zone_idx[i])
+                            fh[:, zone_columns_idx[bus_cols_zone_idx[i]] + 1] -= fh[
+                                :, bus_columns_idx[i] + 1
+                            ]
+                        end
+                    end
+
+                    # check if flexibility in any zone is less than the sum of buses
+                    for i in 1:flexible_zone_num
+                        if any(x -> x < 0, fh[:, zone_columns_idx[flexible_zone_num] + 1])
+                            @error(
+                                "Input ERROR: Total zone flexibility less than sum of bus flexibility for zone " *
+                                    string(flexible_zone_id[i])
+                            )
+                        end
+                    end
+                end
+                # convert zone-level to bus-level
+                converted_zone_flexibility =
+                    Matrix(fh[:, [i + 1 for i in zone_columns_idx]]) * zone_to_bus_shares[zone_cols_idx, :]
+
+                # add bus-level columns on top of converted bus-level flexibility matrix
+                flex_bus_idx = zeros(Int64, 0)
+                for i in sets.bus_idx
+                    bus = case.busid[i]
+                    # add specified flexibility to the corresponding bus
+                    if bus in flexible_bus_id
+                        converted_zone_flexibility[:, i] += fh[string(bus)]
+                    end
+
+                    # identify flexible bus by their total flexibility and append to list
+                    if any(x -> x > 0, converted_zone_flexibility[:, i])
+                        append!(flex_bus_idx, i)
+                    end
+                end
+                # for cost, the zone numbers apply to all buses except those with dedicated columns
+            else
+                # convert zone-level cost using incidence matrix
+                converted_zone_flexibility =
```

Comment:
Should this be `converted_zone_cost`, along with the other instances of `converted_zone_flexibility` in the cost-related section?